File: src/nodejs/src/entities/Order.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, JoinColumn } from "typeorm";
import { BaseEntity } from "./BaseEntity";
import { Patient } from "./Patient";

/**
 * Entity representing an order in the system.
 */
@Entity("order")
export class Order extends BaseEntity {
  /**
   * The unique identifier for the order.
   */
  @PrimaryGeneratedColumn({ type: "bigint", name: "id" })
  id!: number;

  /**
   * The message associated with the order.
   */
  @Column({ type: "varchar", name: "message", length: 500 })
  message!: string;

  /**
   * The ID of the patient who owns this order.
   */
  @Column({ type: "bigint", name: "patient_id" })
  patientId!: number;

  /**
   * The patient who owns this order.
   */
  @ManyToOne(() => Patient, (patient) => patient.orders)
  @JoinColumn({ name: "patient_id" })
  patient!: Patient;
}

File: src/python/app/entities/order.py
"""Order entity model."""

from __future__ import annotations
from app.entities.base import Base
from app.entities.base_entity import BaseEntity
from sqlalchemy import Text, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column, relationship
from typing import List, TYPE_CHECKING

if TYPE_CHECKING:
    from app.entities.patient import Patient


class Order(Base, BaseEntity):
    """Order model."""

    __tablename__ = "order"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    message: Mapped[str] = mapped_column(Text, nullable=False)
    patient_id: Mapped[int] = mapped_column(ForeignKey("patient.id", ondelete="CASCADE"), nullable=False)

    # Relationship
    patient: Mapped["Patient"] = relationship("Patient", back_populates="orders")
